### PR TITLE
404 and 500 pages

### PR DESCRIPTION
**Pull request by develas**

### CODE REVIEW

**Comment by tadatuta** [comment]
I'd prefer styled text instead of jpg image (error pages should be as lightweight as possible and still retina-friendly).

File: .githooks/pre-commit/lint [hunk 1]
```diff
@@ -1 +1,3 @@
+#!/bin/bash
```
Comment (tadatuta) [inline .githooks/pre-commit/lint:1]: 👍

File: common.blocks/error-board/error-board.bemtree.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            elem: 'logo'
+        },
+        {
+            elem: 'description',
+            content:[
+                {
+                    elem: 'title',
+                    tag: 'h1',
+                    content : 'Страница не найдена'
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemtree.js]: Please use `i18n` the same way as in https://github.com/tadatuta/bem-forum/pull/16/files#diff-debffe068b0807942ac1a47903381ca2R1

File: common.blocks/error-board/error-board.bemtree.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return [
+        {
+            elem: 'logo'
+        },
+        {
+            elem: 'description',
+            content:[
+                {
+                    elem: 'title',
+                    tag: 'h1',
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemtree.js]: use BEMHTML to set tags

File: common.blocks/error-board/error-board.css [hunk 4]
```diff
@@ -0,0 +1,18 @@
+.error-board{
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.css]: missing space before `{`

File: common.blocks/error-board/error-board.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    height: 150px;
+    background: url('404_error.jpg') no-repeat;
+    margin: 0 40px 40px 0;
+}
+
+
+.error-board__description {
+    display: inline-block;
+    vertical-align: top;
+}
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.css]: new line at EOF needed

File: common.blocks/error-board/error-board.deps.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+({
+    shouldDeps:
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.deps.js]:
``` js
({
    shouldDeps: ['link']
})
```

should be enough.

File: common.blocks/page/_view/page_view_404.bemtree.js [hunk 7]
```diff
@@ -1,3 +1,12 @@
 block('page').mod('view', '404').content()(function() {
-    return '404';
+
+    return [
+        {
+            block: 'error-board'
+        },
+        {
```
Comment (tadatuta) [inline common.blocks/page/_view/page_view_404.bemtree.js]: I'd move `{ block: 'issues' }` inside `error-board`

File: common.blocks/page/page.deps.js [hunk 8]
```diff
@@ -3,6 +3,9 @@
         {
             mods: { view: ['404'] }
```
Comment (tadatuta) [inline common.blocks/page/page.deps.js]: it's possible to just add `500` to this array:

``` js
mods: { view: ['404', '500'] }
```

File: server/controllers/gh.js [hunk 9]
```diff
@@ -62,6 +62,23 @@ function getComments(req, res) {
     });
 }
 
+function notFoundPage(req, res){
```
Comment (tadatuta) [inline server/controllers/gh.js]: missing space before `{`

File: server/controllers/gh.js [hunk 10]
```diff
@@ -62,6 +62,23 @@ function getComments(req, res) {
...
+    makeIssueRequest(issuesRequestUrl).then(function(issues) {
+        var latestIssues =  issues.filter(function(issue) {
```
Comment (tadatuta) [inline server/controllers/gh.js]: extra space after `=`

File: server/controllers/gh.js [hunk 11]
```diff
@@ -62,6 +62,23 @@ function getComments(req, res) {
...
+                                var tr = Math.abs(moment(issue.created_at).diff(moment(), 'days')) <= 30;
```
Comment (tadatuta) [inline server/controllers/gh.js]: this part can be optimized like this:

``` js
return issue.created_at <= 30 * 24 * 60 * 60 * 1000; // 30 days
```

File: server/index.js [hunk 12]
```diff
@@ -43,6 +46,13 @@ if (isDev) {
     app.use(require('errorhandler')());
 }
 
+function errorHandler(err, req, res, next) {
+    console.log('errorHandler');
```
Comment (tadatuta) [inline server/index.js]: this shouldn't be kept here

File: server/index.js [hunk 13]
```diff
@@ -43,6 +46,13 @@ if (isDev) {
...
+    res.status(500);
+    render(req, res, { view: '500' });
+    next(err); // added this row because eslint says: error  'next' is defined but never used  no-unused-vars
```
Comment (tadatuta) [inline server/index.js]: just put the function inside special comments:

``` js
/*eslint-disable no-unused-vars */
...
/*eslint-enable no-alert */
```

for more info see http://stackoverflow.com/questions/27732209/turning-off-eslint-rule-for-a-specific-line

File: common.blocks/page/_view/page_view_500.bemtree.js [hunk 14]
```diff
@@ -0,0 +1,19 @@
+block('page').mod('view', '500').content()(function() {
```
Comment (tadatuta) [inline common.blocks/page/_view/page_view_500.bemtree.js:1]: it's just about `page` content, so there's no need to provide doctype, html, head and body tags, just content itself:

``` js
block('page').mod('view', '500').content()(function() {
    return [
        {
            tag: 'h1',
            content: 'Внутренняя серверная ошибка'
        },
        {
            tag: 'p',
            content: 'Извините, что-то пошло не так. Сервер не смог обработать ваш запрос.'
        },
        {
            tag: 'p',
            content: 'Пожалуйста, попробуйте еще раз позже.'
        }
    ];
});
```

Also `i18n` should be used.

File: server/index.js [hunk 15]
```diff
@@ -33,7 +34,9 @@ app
     .use(slashes());
 // TODO: csrf, gzip
 
+
 app.use(router);
+app.use(errorHandler);
```
Comment (tadatuta) [inline server/index.js]: it should be moved to line 48, otherwise next part piece of code won't work as expected.

File: server/controllers/gh.js [hunk 16]
```diff
@@ -94,5 +111,6 @@ function makeIssueRequest(issueRequestUrl) {
 module.exports = {
     getIssues,
     getIssue,
-    getComments
+    getComments,
+    notFoundPage
```
Comment (tadatuta) [inline server/controllers/gh.js]: I'd rename it to `get404` for consistency

File: server/controllers/gh.js [hunk 17]
```diff
@@ -62,6 +62,23 @@ function getComments(req, res) {
...
+function notFoundPage(req, res){
+    makeIssueRequest(issuesRequestUrl).then(function(issues) {
+        var latestIssues =  issues.filter(function(issue) {
+                                var tr = Math.abs(moment(issue.created_at).diff(moment(), 'days')) <= 30;
+                                return tr;
+                            });
+
+    res.status(404);
+    res.send(render(req, res, {
+                issues: latestIssues,
```
Comment (tadatuta) [inline server/controllers/gh.js]: something strange with indentation here

File: server/controllers/gh.js (hunk 11 above)
Comment (tadatuta) [inline server/controllers/gh.js]: BTW, don't you think it's better to render just last 10 issues?
there might be 1000 or 0 issue in last 30 days and 10 is always 10 ;)

File: common.blocks/issue/issue.bemtree.js [hunk 18]
```diff
@@ -40,15 +40,15 @@ block('issue').content()(function() {
             elem: 'content',
             content: issue.html
         },
-        this.data.url.pathname === '/' ? {
+        this.data.url.pathname.match(/^\/\d+\/$/g) !== null ? {
```
Comment (tadatuta) [inline common.blocks/issue/issue.bemtree.js]: no need to compare with `null`:

``` js
this.data.url.pathname.match(/^\/\d+\/$/g) ? { } : {}
```

BTW, what's the purpose of this change?

Comment (develas) [inline common.blocks/issue/issue.bemtree.js]: Before this we have used block "Comments" in "Issue" block if url.pathname !== '/'. But we don't need comments(unfolded) on 404 page. 
Right now the unfolded "Comments" will be used only on Issue page, e.g /34/. 
If you have correct decision it will be great to see it :)

Comment (tadatuta) [inline common.blocks/issue/issue.bemtree.js]: I prefer not to use regexps as long as it possible. So I'd better use something like:

``` js
this.data.url.pathname === '/' || this.data.view === '404'
```

File: common.blocks/page/_view/page_view_500.bemtree.js (hunk 14 above)
Comment (develas) [inline common.blocks/page/_view/page_view_500.bemtree.js:1]: Why did you wrote that tags from error-board.bemtree.js should be set in BEMHTML but here they are in BEMTREE? Or for such static page like 505 it is ok?

Comment (tadatuta) [inline common.blocks/page/_view/page_view_500.bemtree.js:1]: It's more about having some BEM entity to apply templates to.
If you have a block or element it's possible to create template for it. But here we don't have any so it's fine to keep it like this.

File: .githooks/pre-commit/lint [hunk 19]
```diff
@@ -1 +0,0 @@
-npm run lint
```
Comment (tadatuta) [inline .githooks/pre-commit/lint:1]: why?

File: common.blocks/error-board/error-board.bemhtml.js [hunk 20]
```diff
@@ -0,0 +1,4 @@
+block('error-board')(
+    elem('title').tag()('h1'),
+    elem('titleIssues').tag()('h2')
+);
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemhtml.js]: new line at EOF (everywhere) needed

File: common.blocks/error-board/error-board.bemhtml.js [hunk 21]
```diff
@@ -0,0 +1,4 @@
+block('error-board')(
+    elem('title').tag()('h1'),
+    elem('titleIssues').tag()('h2')
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemhtml.js]: let's use dash to separate words (`title-issues`)

File: common.blocks/error-board/error-board.bemtree.js [hunk 22]
```diff
@@ -0,0 +1,36 @@
+block('error-board').content()(function() {
+    var block = this.block,
+        i18n = this.require('i18n');
+        console.log();
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemtree.js]: !

File: common.blocks/error-board/error-board.bemtree.js [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemtree.js]: extra empty lines

File: common.blocks/error-board/error-board.bemtree.js [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+    return [
+        {
+            elem: 'description',
+            content:[
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.bemtree.js]: space after `:` needed

File: common.blocks/error-board/error-board.css [hunk 25]
```diff
@@ -0,0 +1,8 @@
+.error-board {
+    font-family: Verdana, Arial;
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.css]: I'd remove this line (font-family is inherited from parents)

File: common.blocks/error-board/error-board.deps.js [hunk 26]
```diff
@@ -0,0 +1,7 @@
+({
+    shouldDeps: 
+        {
+            block: 'link',
+            block: 'issues'
```
Comment (tadatuta) [inline common.blocks/error-board/error-board.deps.js]:
``` js
({
    shouldDeps: ['link', 'issues']
})
```

File: server/controllers/gh.js [hunk 27]
```diff
@@ -62,6 +62,20 @@ function getComments(req, res) {
     });
 }
 
+function get404 (req, res) {
+    makeIssueRequest(issuesRequestUrl).then(function(issues) {
+        var latestIssues = issues.slice(0, 10);
+
+        res.status(404);
+        res.send(render(req, res, {
+                issues: latestIssues,
```
Comment (tadatuta) [inline server/controllers/gh.js]: extra indentation

File: server/index.js [hunk 28]
```diff
@@ -33,8 +34,10 @@ app
     .use(slashes());
 // TODO: csrf, gzip
 
+
```
Comment (tadatuta) [inline server/index.js]: extra new line here and at 40
